app/services/my_anime_list_sync_service.rb

```
@@ -0,0 +1,203 @@
+    )
+
+    request_status(request) do |response|
+      yield response
+    end
+    request.run
+  end
+
+  def post(url, profile, body)
+    request = Typhoeus::Request.new(
+      build_url(url),
+      method: :post,
+      userpwd: simple_auth(profile),
+      body: body
+    )
+
+    # TODO: @nuck is there a better way to handle this?
+    request_status(request) do |response|
+      # doesn't need to do anything if success
+    end
+    request.run
+  end
+
+  def put(url, profile, body)
+    request = Typhoeus::Request.new(
+      build_url(url),
+      method: :put,
+      userpwd: simple_auth(profile),
+      body: body
+    )
+
+    # TODO: @nuck is there a better way to handle this?
+    request_status(request) do |response|
+      # doesn't need to do anything if success
+    end
+    request.run
+  end
+
+  def delete(url, profile)
+    request = Typhoeus::Request.new(
+      build_url(url),
+      method: :delete,
+      userpwd: simple_auth(profile)
+    )
+
+    # TODO: @nuck is there a better way to handle this?
+    request_status(request) do |response|
+      # doesn't need to do anything if success
+    end
+    request.run
+  end
+
+  def request_status(request)
+    # will return request or nil
+    request.on_complete do |response|
+      if response.success?
+        # this is being sent up to either
+        # get/create/update/delete
+        # afterwards the chosen method will send it up
+        # to the parent request under my_anime_list.rb
+
+        # delete will not have a body so need try catch
+        yield response&.body
+      elsif response.timed_out?
+        # aw hell no
+        $stderr.puts('got a time out')
+      elsif response.code.zero?
+        # Could not get an http response, something's wrong.
+        $stderr.puts(response.return_message)
+      else
+        # Received a non-successful http response.
+        $stderr.puts('HTTP request failed: ' + response.code.to_s)
+      end
+    end
+  end
```

It's probably better to raise an exception when we get a non-200 response so that it gets caught by Sentry and emailed to me ;)